src/dashboard/src/components/administration/views.py

```diff
@@ -223,6 +229,146 @@ def storage(request):
     system_directory_description = 'Available storage'
     return render(request, 'administration/locations.html', locals())
 
+def usage(request):
+    usage_dirs = _usage_dirs()
+
+    context = {'usage_dirs': usage_dirs}
+    return render(request, 'administration/usage.html', context)
+
+def _usage_dirs(calculate_usage=True):
+    # Put spaces before directories contained by the spaces
+    #
+    # Description is optional, but either a path or a location purpose (used to
+    # look up the path) should be specified
+    #
+    # If only certain sudirectories within a path should be deleted, set
+    # 'subdirectories' to a list of them
+    dir_defs = (
+        ('shared', {
+            'path': helpers.get_client_config_value('sharedDirectoryMounted')
+        }),
+        ('dips', {
+            'description': 'DIP uploads',
+            'path': os.path.join('watchedDirectories', 'uploadedDIPs'),
+            'contained_by': 'shared'
+        }),
+        ('rejected', {
+            'description': 'Rejected',
+            'path': 'rejected',
+            'contained_by': 'shared'
+        }),
+        ('failed', {
+            'description': 'Failed',
+            'path': 'failed',
+            'contained_by': 'shared'
+        }),
+        ('tmp', {
+            'description': 'Temporary file storage',
+            'path': 'tmp',
+            'contained_by': 'shared'
+        })
+    )
+
+    dirs = collections.OrderedDict(dir_defs)
+
+    # Resolve location paths and make relative paths absolute
+    for _, dir_spec in dirs.iteritems():
+        if 'contained_by' in dir_spec:
```

Comment:
I do the contained_by thing, rather than always path.join'ing the shared directory, in case in the future we want to support having things on multiple volumes, etc. Maybe it's overkill for now?